env/env_posix.cc

```diff
@@ -509,9 +513,17 @@ Env* Env::Default() {
   return &default_env;
 }
 
+std::shared_ptr<Env> GetCompositeBaseEnv() {
+  static std::shared_ptr<PosixEnv> shared_env;
+  // Call Env::Default to make sure the thread singletons have been setup
+  (void)Env::Default();
+  if (!shared_env) shared_env.reset(new PosixEnv());
```

Comment:
I had to create a new PosixEnv instance(along with new thread pools) to allow it to be destroyed. The destruction was needed to join the threads before the composite env was destroyed. Since Env::Default is not reference counted it can't be destroyed safely.